modules/ROOT/pages/as2/as2-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,257 @@
+	</tls:context>
+
+<!-- Receive global config -->
+	<as2-mule4:listener-config name="AS2_HTTPS_Listener_config" httpListenerConfig="HTTPS_Listener_config">
+		<as2-mule4:async-mdn-connection-provider-connection tlsContext="TLS_Context"/>
+		<as2-mule4:self-config as2Name="selfId" x509Alias="selfAlias" email="self@email.com"/>
+		<as2-mule4:partner-config as2Name="partnerId" x509Alias="partnerAlias" email="partner@email.com"/>
+		<as2-mule4:key-store-config keystorePassword="keystorepwd" keystorePath="path/to/keystore"/>
+	</as2-mule4:listener-config>
+
+<!-- Receive MDN global config -->
+	<as2-mule4:mdn-listener-config name="AS2_Connector_HTTPS_Mdn_listener_config" httpListenerConfig="HTTPS_Listener_config">
+		<as2-mule4:self-config as2Name="selfId" x509Alias="selfAlias" email="self@email.com"/>
+		<as2-mule4:partner-config as2Name="partnerId" x509Alias="partnerAlias" email="partner@email.com"/>
+		<as2-mule4:key-store-config keystorePassword="keystorepwd" keystorePath="path/to/keystore"/>
+	</as2-mule4:mdn-listener-config>
+```
+----
+
+==== Receive source
+
+The receive source needs to define the `path` of the listening endpoint.
+
+* Before:
+[source,xml,linenums]
+----
+```
+<as2:receive doc:name="Receive" config-ref="AS2_Config"/>
+```
+----
+
+* After:
+[source,xml,linenums]
+----
+```
+<flow name="AS2_HTTPS_Listener_Flow">
+    <as2-mule4:as2-listener path="/toSelf" config-ref="AS2_HTTPS_Listener_config"/>
+</flow>
+```
+----
+
+==== Send AS2 messages
+
+The send operation is divided into 2 separate operations - `send-with-async-mdn` and `send-with-sync-mdn`
+
+* Before:
+[source,xml,linenums]
+----
+```
+<as2:send timeout="send timeout" config-ref="AS2_Config" input="Hello world" fileName="fileName"/>
+```
+----
+
+* After
+[source,xml,linenums]
+----
+
+```
+<as2-mule4:send-with-async-mdn  fileName="fileName" config-ref="AS2_Connector_Send_Sync_config" mimeType="application/xml" requireProcessed="false">
+    <as2-mule4:content ><![CDATA[#['Hello world']]]></as2-mule4:content>
+</as2-mule4:send-with-async-mdn>
+
+<!-- OR -->
+
+<as2-mule4:send-with-sync-mdn  fileName="fileName" config-ref="AS2_Connector_Send_Sync_config" mimeType="application/xml" requireProcessed="false">
+    <as2-mule4:content ><![CDATA[#['Hello world']]]></as2-mule4:content>
+</as2-mule4:send-with-sync-mdn>
+
+```
+----
+
+==== Receive MDN source
+
+The receive MDN flow needs to be defined if you are using `send-with-async-mdn`.
+
+* Before:
+```
+no config needed
+```
+
+* After:
+[source,xml,linenums]
+----
+```
+<flow name="AS2_HTTPS_MDN_Listener_Flow">
+    <as2-mule4:as2-mdn-listener config-ref="AS2_Connector_HTTPS_Mdn_listener_config" path="/mdn"/>
+</flow>
+`
+----
+
+
+=== Post Upgrade Steps
+
+After you install the latest version of the connector, follow these steps to complete the upgrade:
+
+. Notify Partners of the new URL/port configuration that corresponds to the HTTP listener you are using.
+
+. Verify that the connector is working by ???
```

Comment:
Verify that the connector is working by making sure there are no exceptions on startup and testing against the AS2 partner.